sessionctx/variable/session.go

```diff
@@ -812,6 +816,9 @@ func (s *SessionVars) SetSystemVar(name string, val string) error {
 	case MaxExecutionTime:
 		timeoutMS := tidbOptPositiveInt32(val, 0)
 		s.MaxExecutionTime = uint64(timeoutMS)
+	case InnodbLockWaitTimeout:
+		lockWaitSec := tidbOptInt64(val, DefInnodbLockWaitTimeout)
```

Comment:
We need to validate the range.

Reply:
this is done in function `ValidateSetSystemVar`
```
case InnodbLockWaitTimeout:
	return checkUInt64SystemVar(name, value, 1, 1073741824, vars)
```

```
mysql> set global innodb_lock_wait_timeout = 0;
Query OK, 0 rows affected, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+---------------------------------------------------------+
| Level   | Code | Message                                                 |
+---------+------+---------------------------------------------------------+
| Warning | 1292 | Truncated incorrect innodb_lock_wait_timeout value: '0' |
+---------+------+---------------------------------------------------------+
1 row in set (0.00 sec)
```